extensions/iidm/src/MeasurementsXmlSerializer.cpp

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <powsybl/iidm/extensions/iidm/MeasurementsXmlSerializer.hpp>
+
+#include <powsybl/iidm/Enum.hpp>
+#include <powsybl/iidm/Identifiable.hpp>
+#include <powsybl/iidm/converter/Constants.hpp>
+#include <powsybl/iidm/converter/xml/NetworkXmlReaderContext.hpp>
+#include <powsybl/iidm/converter/xml/NetworkXmlWriterContext.hpp>
+#include <powsybl/iidm/extensions/iidm/Measurement.hpp>
+#include <powsybl/iidm/extensions/iidm/MeasurementAdder.hpp>
+#include <powsybl/iidm/extensions/iidm/Measurements.hpp>
+#include <powsybl/iidm/extensions/iidm/MeasurementsAdder.hpp>
+#include <powsybl/xml/XmlStreamReader.hpp>
+#include <powsybl/xml/XmlStreamWriter.hpp>
+
+namespace powsybl {
+
+namespace iidm {
+
+namespace extensions {
+
+namespace iidm {
+
+MeasurementsXmlSerializer::MeasurementsXmlSerializer() :
+    converter::xml::AbstractExtensionXmlSerializer("measurements", "network", "m", "http://www.powsybl.org/schema/iidm/ext/measurements/1_0") {
+}
+
+Extension& MeasurementsXmlSerializer::read(Extendable& extendable, converter::xml::NetworkXmlReaderContext& context) const {
+    if (!stdcxx::isInstanceOf<Identifiable>(extendable)) {
+        throw AssertionError(stdcxx::format("Unexpected extendable type: %1% (%2% expected)", stdcxx::demangle(extendable), stdcxx::demangle<Identifiable>()));
+    }
+    extendable.newExtension<MeasurementsAdder>().add();
+    auto& measurements = extendable.getExtension<Measurements>();
+    context.getReader().readUntilEndElement("measurements", [&measurements, &context]() {
+        if (context.getReader().getLocalName() == "measurement") {
+            MeasurementAdder adder = measurements.newMeasurement()
+                .setId(context.getReader().getAttributeValue(converter::ID))
```

Comment:
To be discussed: should be share the IIDM constant with extensions or create a dedicated set of constants. Same remark for the other serializer class

Reply:
I think an extension serializer should probably create its own constants to avoid unwanted side effect if core constants change..

Reply:
Agree!